src/devices/Rtc/Devices/Ds3231/Ds3231Alarm1.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Iot.Device.Rtc
+{
+    /// <summary>
+    /// Represents alarm 1 on the DS3231
+    /// </summary>
+    public class Ds3231Alarm1
+    {
+        /// <summary>
+        /// Day of month or day of week of the alarm. Which one it is depends on the match mode
+        /// </summary>
+        public int DayOfMonthOrWeek { get; set; }
```

Comment:
Why not use a TimeStamp for the Hour, Days, etc? 
That would save you a lot of checks.